src/pumipic_adjacency.tpp

```diff
@@ -395,16 +395,25 @@ namespace pumipic {
     printf("Min area is: %.15f, Planned tol is %.15f\n", min_area, tol);
     return tol;
   }
-  
-  template <class ParticleType, typename Segment3d, typename SegmentInt>
-  bool search_mesh(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
+
+  template <class ParticleType, typename Segment3d, typename SegmentInt, typename Func>
+  bool particle_search(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
                    Segment3d x_ps_orig, Segment3d x_ps_tgt, SegmentInt pids,
                    o::Write<o::LO>& elem_ids,
                    bool requireIntersection,
                    o::Write<o::LO>& inter_faces,
                    o::Write<o::Real>& inter_points,
```

Comment:
I think `inter_faces` and `inter_points` are specific for a specific choice of behavior at the element boundaries. These should be inside the functor, not in the function signature here.

Reply:
These are coming from the search and are the faces and points where the particles intersect.